tests/random/prng_test.go

```diff
@@ -59,116 +118,312 @@ func TestRead(t *testing.T) {
 			var seed [32]byte
 			copy(seed[:], crypto.Keccak256([]byte(tt.seedFrom)))
 
-			samples, state, err := prng.Sample(nil, seed, uint16(tt.bitsPerSample), uint16(tt.numSamples))
-			if err != nil {
-				t.Fatalf("Sample(%x, 8 bits, 1000 samples) error %v", seed, err)
-			}
+			for _, ii := range getImplementations(t) {
+				t.Run(ii.name, func(t *testing.T) {
 
-			t.Run("probabilistic tests", func(t *testing.T) {
-				var min, max, sum uint64
-				min = math.MaxUint64
-				for i, s := range samples {
-					if !s.IsUint64() {
-						t.Fatalf("sample[%d].IsUint64() got false; want true", i)
+					samples, err := ii.prng.Sample(nil, seed, uint16(tt.bitsPerSample), uint16(tt.numSamples))
+					if err != nil {
+						t.Fatalf("Sample(%x, 8 bits, 1000 samples) error %v", seed, err)
 					}
 
-					u := s.Uint64()
-					sum += u
-					if u < min {
-						min = u
-					}
-					if u > max {
-						max = u
+					var min, max, sum uint64
+					var onesCount int
+
+					numMax := (1 << tt.bitsPerSample) - 1
+					distMean := float64(numMax) / 2.
+					distVar := float64(numMax*numMax) / 12.
```

Comment:
These are only used down in `t.Run("Sum"` so please declare them there instead.

Also, where does the 12 come from? Please add a comment to explain

Reply:
This is the sample variance of a discrete uniform distribution. There was a small error in the formula - thanks for making me revisit it.